File: src/domain/RPSRound.java
package domain;

public class RPSRound {
    private final RPSHand opponent;
    private final RPSHand player;

    public RPSRound(String opponent, String result) {
        this.opponent = RPSHand.getMatching(opponent);
        this.player = this.opponent.getHandFor(result);
    }

    public int getPoints() {
        return player.getPoints() + player.getPointsAgainst(opponent);
    }
}

File: Day1/src/domain/RPSRound.kt
package domain

class RPSRound(opponent: String?, result: String?) {
    private val opponent: RPSHand
    private val player: RPSHand

    init {
        this.opponent = RPSHand.getMatching(opponent)
        player = this.opponent.getHandFor(result)
    }

    val points: Int
        get() = player.points + player.getPointsAgainst(opponent)
}